common/common.cpp

```diff
@@ -1122,34 +1123,50 @@ std::vector<std::string> string_split(std::string input, char separator) {
     return parts;
 }
 
-std::vector<llama_sampler_type> sampler_types_from_names(const std::vector<std::string> & names) {
+std::vector<llama_sampler_type> sampler_types_from_names(const std::vector<std::string> & names, bool allow_alt_names) {
+    std::unordered_map<std::string, llama_sampler_type> sampler_canonical_name_map {
+        {"top_k",       llama_sampler_type::TOP_K},
+        {"top_p",       llama_sampler_type::TOP_P},
+        {"typical_p",   llama_sampler_type::TYPICAL_P},
+        {"min_p",       llama_sampler_type::MIN_P},
+        {"tfs_z",       llama_sampler_type::TFS_Z},
+        {"temperature", llama_sampler_type::TEMPERATURE}
+    };
+
     // since samplers names are written multiple ways
     // make it ready for both system names and input names
-    std::unordered_map<std::string, llama_sampler_type> sampler_name_map {
-        {"top_k",       llama_sampler_type::TOP_K},
+    std::unordered_map<std::string, llama_sampler_type> sampler_alt_name_map {
```

Comment:
Do we actually need this name / alt-name separation?
I don't see any drawbacks the way it was before

Reply:
It's still the way it was before ...for `main`. This separation is only for the `server`. For `main` the alt names are sort of understandable, because the user will type them by hand. But API is mostly computer-to-computer communication.

And the thing is that supporting alt names for `server` will require an additional effort and increasing the complexity. In my first (not-submitted) draft I actually tried to add the support for alt names to the API, but it really complicated things. The problem with alt names is that you can no longer rely on `sampler_type_to_name_string` to get the sampler name for `get_formated_generation`, and all the raw input names must be stored alongside their parsed versions. And they also need to be filtered (I guess), so the wrong names won't show in `get_formated_generation`. This means that `sampler_types_from_names` will also need to be completely reworked. And all the alt names then need to be documented, also slightly complicating the specs. In the end I decided that the ends don't justify the means and separated the "canonical" names from the rest.

Of course, if needed, the support for alt names for `server` can be added later, since it won't break the backwards compatibility.